### PR TITLE
Make spec repr generic

### PR DESCRIPTION
**Pull request by jacobtomlinson**
Set `SpecCluster` repr to use type for class name. Useful when subclassing.

### CODE REVIEW

**Comment by mrocklin** [comment]
+1 from me.  I'll suggest the following test:

```python
async def test_repr():
    class MyCluster(SpecCluster):
        pass

    async with MyCluster(..., asynchronous=True) as cluster:
        assert "MyCluster" in str(cluster)
```

File: distributed/deploy/tests/test_spec_cluster.py [hunk 1]
```diff
@@ -81,6 +81,15 @@ def test_loop_started():
     )
 
 
+@pytest.mark.asyncio
+async def test_repr():
+    class MyCluster(SpecCluster):
+        pass
+
+    async with MyCluster(..., asynchronous=True) as cluster:
```
Comment (mrocklin) [inline distributed/deploy/tests/test_spec_cluster.py]: Ah, blind copy-paste issue I think?  I used `...` to mean "I haven't thought about what should go in here too much, but you could probably copy-paste from similar tests"

Comment (jacobtomlinson) [reply distributed/deploy/tests/test_spec_cluster.py]: Ha my bad! I assumed that was some clever shorthand for defaulting everything that I hadn't seen before.

**Comment by mrocklin** [comment]
Thanks @jacobtomlinson !

**Pull request merged**